src/gui/shell.cpp

```diff
@@ -365,6 +368,15 @@ void Shell::init()
 
 	// Set initial value
 	m_nvim->api0()->vim_set_var("GuiWindowFrameless", (windowFlags() & Qt::FramelessWindowHint) ? 1: 0);
+
+	// Make the shell visible even when default_colors_set is not received,
+	// e.g. when using the deprecated cell-based grid protocol.
+	QTimer::singleShot(888, [&]() {
```

Comment:
Let's move the hard-coded `888` constant to a named variable.

Something along the lines of:
`const int timeoutDefaultColorEventNotReceivedMSec { 750 };`

Reply:
I reworked things a bit and this ended up getting called `constexpr int timeout_visibility{ 850 };` mostly since it avoided too long of a line and because it looked a little more consistent alongside the other variables in scope.

The timeout was kept at 850 just to keep the chances higher of it firing after the first event, e.g. on slower hardware or when launching nvim + nvim-qt binaries over slow storage such as NFS. I hope that's cool.

I figured out that this was the snippet I had to add to `~/.config/nvim-qt/nvim-qt.conf` to enable the old protocol.

```
[General]
ext_linegrid=false
```

Once I did that I tested it and as far as I can tell the `default_colors_set` message is still received and the widget is still made visible (even without the timer). I still like the peace of mind of having the timer, though.

The blink is still seen with my nvim config w/ ext_linegrid=false, but that's fine IMO since it's no worse than before and only an issue when using the old protocol.

When using `nvim-qt -- -u NONE` the old protocol does seem to mostly avoid the blink, possibly due to a different sequence of commands getting received. Anyways, that's just my testing w/ the two nvim versions I tried. I'm curious to hear if you see similar results in your test.

Thanks for the review ~ I'm more than happy to tweak things further if you have other preferences.